src/Controller/FigureController.php

```diff
@@ -61,6 +61,46 @@ public function index(Request $request, FigureManagerInterface $figureManager, S
         ]);
     }
 
+    #[Route('/new', name: 'figure_new')]
+    #[Route('/figure/edit/{slug}', name: 'figure_edit')]
+    public function new(SnowFigure $figure = null, Request $request, FigureManagerInterface $figureManager): Response
+    {
+        $this->denyAccessUnlessGranted('IS_AUTHENTICATED_FULLY');
+
+        if (!$figure) {
+            $figure = new SnowFigure();
+        }
+
+        $user = $this->getUser();
+        $form = $this->createForm(FigureFormType::class, $figure);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            try {
+                if (!$figure->getId()) {
```

Comment:
Tu pourrais gérer l'edition ou la création dans le manager et placer ton if dans ton manager. Ca allegera la lecture de ton controller.
```suggestion
                $figureManager->handleFigure($figure);
```

Reply:
Done